src/wrapped_radix_engine_toolkit/low_level/notarized_transaction.rs

```diff
@@ -0,0 +1,122 @@
+use crate::prelude::*;
+
+#[derive(Debug, Clone, Eq, PartialEq, uniffi::Record)]
+pub struct NotarizedTransaction {
+    pub signed_intent: SignedIntent,
+    pub notary_signature: NotarySignature,
+}
+
+impl NotarizedTransaction {
+    pub fn new(
+        signed_intent: SignedIntent,
+        notary_signature: NotarySignature,
+    ) -> Self {
+        Self {
+            signed_intent,
+            notary_signature,
+        }
+    }
+
+    pub fn compile(&self) -> CompiledNotarizedIntent {
+        let scrypto: ScryptoNotarizedTransaction = self.clone().into();
+        let bytes: BagOfBytes = RET_compile_notarized_tx(&scrypto)
+            .expect("Should always be able to compile a notarized tx")
```

Comment:
@0xOmarA is it OK to unwrap `compile` of NotarizedTransaction? When can it fail?